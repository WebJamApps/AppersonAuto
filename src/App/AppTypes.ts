export interface GoogleBody {
  clientId?: string,
  redirectUri: string,
  code: string,
  state(): string,
}
