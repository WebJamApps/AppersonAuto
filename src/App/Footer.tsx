import React from 'react';

const footerLinks = (): JSX.Element => (
  <>
    <div style={{ textAlign: 'center', padding: '6px' }} className="footerLinks">
      <a href="https://www.facebook.com/pages/Apperson-Automotive/361511727202422">
        <i style={{ color: '#45c9ff' }} className="fab fa-facebook-square fa-2x" aria-hidden="true" />
      </a>
      <a href="https://www.yellowpages.com/salem-va/mip/apperson-automotive-482421530" style={{ marginLeft: '40px' }}>
        <img
          style={{ marginBottom: '15px', verticalAlign: 'middle' }}
          src="https://insightmg.com/wp-content/uploads/2015/12/YP_Insight-Marketing-Group_Marketing-for-Medical-Practices.png"
          alt="yellowpages"
          width="40px"
        />
      </a>
      <a href="https://www.yelp.com/biz/apperson-automotive-salem" style={{ marginLeft: '30px' }}>
        <img
          style={{ marginBottom: '15px', verticalAlign: 'middle' }}
          src="https://dl.dropboxusercontent.com/s/pb5qn5wyps6k56t/yelp-logo-icon-7.jpg?dl=0"
          alt="yelp"
          width="40px"
        />
      </a>
    </div>
    <div style={{ textAlign: 'center', padding: '6px' }}>
      <span className="nohover">
        Powered by &nbsp;
      </span>
      <a className="wjllc" target="_blank" rel="noopener noreferrer" href="https://www.web-jam.com">Web Jam LLC</a>
    </div>
  </>
);

const Footer = (): JSX.Element => (
  <div
    id="wjfooter"
    className="footer"
    style={{
      marginTop: '20px', paddingTop: '20px', paddingBottom: '20px', bottom: '0',
    }}
  >
    { footerLinks() }
  </div>
);

export default Footer;
