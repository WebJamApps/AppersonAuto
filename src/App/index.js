import 'core-js/stable';
import 'regenerator-runtime/runtime';
import React, { PureComponent } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import AutoMaint from '../containers/AutoMaintenance';
import GeneralAutoRepair from '../containers/GeneralAutoRepair';
import MajorAutoRepair from '../containers/MajorAutoRepair';
import DefaultContact from '../containers/Contact';
import AppFourOhFour from './404';
import AppTemplateDefault from './AppTemplate';
import DefaultHome from '../containers/Homepage';
import mapStoreToProps from '../redux/mapStoreToProps';

export class App extends PureComponent {
  render() {
    return (
      <div id="App" className="App">
        <Router>
          <AppTemplateDefault id="homepage">
            <Switch>
              <Route exact path="/" component={DefaultHome} />
              <Route path="/auto-maintenance" component={AutoMaint} />
              <Route path="/general-auto-repair" component={GeneralAutoRepair} />
              <Route path="/major-auto-repair" component={MajorAutoRepair} />
              <Route path="/contact" component={DefaultContact} />
              <Route component={AppFourOhFour} />
            </Switch>
          </AppTemplateDefault>
        </Router>
      </div>

    );
  }
}

export default connect(mapStoreToProps, null)(App);
