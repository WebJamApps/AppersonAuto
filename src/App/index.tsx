/* eslint-disable react/destructuring-assignment */
import 'core-js/stable';
import 'regenerator-runtime/runtime';
import React, { Dispatch, PureComponent } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import AutoMaint from '../containers/AutoMaintenance';
import GeneralAutoRepair from '../containers/GeneralAutoRepair';
import MajorAutoRepair from '../containers/MajorAutoRepair';
import DefaultContact from '../containers/Contact';
import AppFourOhFour from './404';
import AppTemplateDefault from './AppTemplate';
import DefaultHome from '../containers/Homepage';

export interface AppProps {
  dispatch: Dispatch<unknown>;
}

export class App extends PureComponent<AppProps> {
  static defaultProps = {
    dispatch: /* istanbul ignore next */(): void => { },
    auth: {
      isAuthenticated: false, token: '', error: '', email: '', user: { userType: '' },
    },
  };

  render(): JSX.Element {
    return (
      <React.StrictMode>
        <div id="App" className="App">
          <Router>
            <AppTemplateDefault dispatch={this.props.dispatch}>
              <Switch>
                <Route exact path="/" component={DefaultHome} />
                <Route path="/auto-maintenance" component={AutoMaint} />
                <Route path="/general-auto-repair" component={GeneralAutoRepair} />
                <Route path="/major-auto-repair" component={MajorAutoRepair} />
                <Route path="/contact" component={DefaultContact} />
                <Route component={AppFourOhFour} />
              </Switch>
            </AppTemplateDefault>
          </Router>
        </div>
      </React.StrictMode>
    );
  }
}

export default App;
