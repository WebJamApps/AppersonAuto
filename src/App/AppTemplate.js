import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import ReactResizeDetector from 'react-resize-detector';
import { GoogleLogin, GoogleLogout } from 'react-google-login';
import { connect } from 'react-redux';
import authUtils from './authUtils';
import mapStoreToProps from '../redux/mapStoreToProps';
import Footer from './Footer';
import menuUtils from './menuUtils';
import menuItems from './menuItems.json';

export class AppTemplate extends Component {
  constructor(props) {
    super(props);
    this.menus = menuItems.menus;
    this.menuUtils = menuUtils;
    this.children = props.children;
    this.state = { menuOpen: false, width: 320 };
    this.close = this.close.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.handleKeyMenu = this.handleKeyMenu.bind(this);
    this.toggleMobileMenu = this.toggleMobileMenu.bind(this);
    this.navLinks = this.navLinks.bind(this);
    this.responseGoogleLogin = this.responseGoogleLogin.bind(this);
    this.responseGoogleLogout = this.responseGoogleLogout.bind(this);
    this.googleButtons = this.googleButtons.bind(this);
    this.authUtils = authUtils;
    this.parentRef = React.createRef();
    this.onResize = this.onResize.bind(this);
  }

  // componentDidMount(){
  //   this.setState
  // }

  onResize(width) { this.setState({ width }); }

  get currentStyles() {
    let result = {};
    this.style = 'wj';
    result = {
      headerImagePath: '../static/imgs/webjamicon7.png',
      headerText1: 'Web Jam LLC',
      headerClass: 'home-header',
      headerImageClass: 'home-header-image',
      sidebarClass: 'home-sidebar',
      menuToggleClass: 'home-menu-toggle',
    };
    result.sidebarImagePath = 'https://upload.wikimedia.org/wikipedia/commons/thumb/a/ab/Lutherrose.svg/800px-Lutherrose.svg.png';
    return result;
  }

  toggleMobileMenu() {
    const { menuOpen } = this.state;
    const mO = !menuOpen;
    this.setState({ menuOpen: mO });
  }

  responseGoogleLogin(response) { return this.authUtils.responseGoogleLogin(response, this); }

  responseGoogleLogout(response) { const { dispatch } = this.props; return this.authUtils.responseGoogleLogout(response, dispatch); }

  close(e) {
    this.setState({ menuOpen: false });
    if (e.target.classList.contains('loginGoogle')) return this.loginGoogle();
    return true;
  }

  handleKeyPress(e) {
    if (e.key === 'Escape') return this.setState({ menuOpen: false });
    return null;
  }

  handleKeyMenu(e) {
    if (e.key === 'Enter') return this.toggleMobileMenu();
    return null;
  }

  googleButtons(type, index) {
    const cId = process.env.GoogleClientId;
    if (type === 'login') {
      return (
        <div key={index} className="menu-item googleLogin">
          <GoogleLogin
            responseType="code"
            clientId={cId}
            buttonText="Login"
            onSuccess={this.responseGoogleLogin}
            onFailure={this.authUtils.responseGoogleFailLogin}
            cookiePolicy="single_host_origin"
          />
        </div>
      );
    } return (
      <div key={index} className="menu-item googleLogout">
        <GoogleLogout clientId={cId} buttonText="Logout" onLogoutSuccess={this.responseGoogleLogout} cookiePolicy="single_host_origin" />
      </div>
    );
  }

  makeMenuLink(menu, index) {
    return (
      <div key={index} className="menu-item">
        <Link to={menu.link} className="nav-link" onClick={this.close}>
          <i className={`${menu.iconClass}`} />
        &nbsp;
          <span className="nav-item">{menu.name}</span>
        </Link>
      </div>
    );
  }

  navLinks() {
    return (
      <div className="nav-list">
        <p style={{ fontSize: '1px', marginBottom: '2px' }} />
        <div className="menu-item" style={{ backgroundColor: '#244a8bff' }}>
          <p style={{ color: '#fff', marginBottom: '2px' }}>
            <a href="http://bit.ly/CollegeLutheranDirections" className="menu-hover" style={{ color: '#88c1ff' }}>
              <span>210 S. College Ave</span>
            </a>
            <br />
            Salem, VA 24153
          </p>
        </div>
        <div className="menu-item" style={{ backgroundColor: '#244a8bff' }}>
          <p style={{ color: '#fff', marginBottom: '2px' }}>
            <span>ph: </span>
            <a href="tel:5403894963" className="menu-hover" style={{ color: '#88c1ff' }}>(540) 389-4963</a>
            <br />
            <span>fax: </span>
            <a href="tel:5403894980" className="menu-hover" style={{ color: '#88c1ff' }}>(540) 389-4980</a>
            <br />
            <a style={{ color: '#88c1ff', wordWrap: 'break-word' }} href="mailto:office1@collegelutheran.org">
              <span className="menu-hover">office1@collegelutheran.org</span>
            </a>
          </p>
        </div>
        {this.menus.map((menu, index) => (this.menuUtils.menuItem(menu, index, this)))}
      </div>
    );
  }

  headerSection(logoWidth, marginTop) {
    return (
      <div id="header" className={`material-header ${this.currentStyles.headerClass}`}>
        <div className="headercontent" />
        <div>
          <div style={{ marginLeft: '5px', marginTop }}>
            <img
              width={logoWidth}
              id="ig"
              className="style-scope iron-image"
              alt="headerlogo"
              src="https://dl.dropboxusercontent.com/s/qg69q7f5so4aqsb/appAutoLogo.png?dl=0"
            />
          </div>
        </div>
      </div>
    );
  }

  render() {
    let logoWidth = '742px', marginTop = '-15px';
    const { menuOpen, width } = this.state;
    console.log(width);
    if (width < 970) { logoWidth = '272px'; marginTop = '1px'; }
    const style = `${this.currentStyles.sidebarClass} ${menuOpen ? 'open' : 'close'}`;
    return (
      <div className="page-host">
        <div tabIndex={0} role="button" id="sidebar" onClick={this.close} onKeyPress={this.handleKeyPress} className={`${style} drawer-container`}>
          <div className="drawer" style={{ backgroundColor: '#c0c0c0' }}>
            <div className="material-header x-scope paper-material-0 drawer" elevation="0" style={{ backgroundColor: '#881204' }}>
              <a style={{ display: 'block', color: '#fff', textDecoration: 'none' }} href="tel:5404447337">
                <h4
                  className="material-header-h4-call"
                  style={{
                    textAlign: 'center', marginRight: '280px', paddingTop: '4px', paddingBottom: 0, width: '200px',
                  }}
                >
                  Call
                  <br />
                  540-444-7337
                </h4>
              </a>
            </div>
            { this.navLinks() }
          </div>
        </div>
        <div className="main-panel">
          <span onClick={this.toggleMobileMenu} onKeyPress={this.handleKeyMenu} id="mobilemenutoggle" tabIndex={0} role="button">
            <i className="fas fa-bars" />
          </span>
          <div className="mainPanel">
            <div className="swipe-area" />
            {this.headerSection(logoWidth, marginTop)}
            <div style={{ width: 'auto' }} id="contentBlock" className="content-block">
              { this.children }
              <Footer />
            </div>
          </div>
        </div>
        <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} targetDomEl={this.parentRef.current} />
      </div>
    );
  }
}
/* istanbul ignore next */
AppTemplate.defaultProps = {
  dispatch: () => {}, auth: { isAuthenticated: false, user: { userType: '' } },
};

AppTemplate.propTypes = {
  location: PropTypes.shape({ pathname: PropTypes.string }).isRequired,
  auth: PropTypes.shape({
    isAuthenticated: PropTypes.bool,
    user: PropTypes.shape({ userType: PropTypes.string }),
  }),
  dispatch: PropTypes.func,
  children: PropTypes.element.isRequired,
};
export default withRouter(connect(mapStoreToProps, null)(AppTemplate));
