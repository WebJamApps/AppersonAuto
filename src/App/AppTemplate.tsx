// @ts-nocheck
import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { withResizeDetector } from 'react-resize-detector';
import { connect } from 'react-redux';
import authUtils from './authUtils';
import mapStoreToProps from '../redux/mapStoreToProps';
import Footer from './Footer';
import MenuItems, { ImenuItem } from './menuItems';

export interface AppTemplateProps {
  targetRef: RefObject<HTMLDivElement>;
  width: number;
  height: number;
  auth: any;
  dispatch: any;
}

export class AppTemplate extends Component<AppTemplateProps> {
  authenticate: any;

  constructor(props: AppTemplateProps) {
    super(props);
    this.menus = MenuItems.menus;
    this.children = props.children;
    this.state = { menuOpen: false };
    this.close = this.close.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.handleKeyMenu = this.handleKeyMenu.bind(this);
    this.toggleMobileMenu = this.toggleMobileMenu.bind(this);
    this.navLinks = this.navLinks.bind(this);
    this.authUtils = authUtils;
  }

  handleKeyPress(e: { key: string; }): (void | null) {
    if (e.key === 'Escape') return this.setState({ menuOpen: false });
    return null;
  }

  handleKeyMenu(e: { key: string; }): (void | null) {
    if (e.key === 'Enter') return this.toggleMobileMenu();
    return null;
  }

  toggleMobileMenu(): void {
    const { menuOpen } = this.state;
    const mO = !menuOpen;
    this.setState({ menuOpen: mO });
  }

  close(): boolean {
    this.setState({ menuOpen: false });
    return true;
  }

  makeMenuLink(menu: ImenuItem, index: number): JSX.Element {
    return (
      <div key={index} className="menu-item">
        <Link to={menu.link} className="nav-link" onClick={this.close}>
          <i className={`${menu.iconClass}`} />
        &nbsp;
          <span className="nav-item">{menu.name}</span>
        </Link>
      </div>
    );
  }

  addressBlock(): JSX.Element { // eslint-disable-line class-methods-use-this
    return (
      <div className="menu-item" style={{ backgroundColor: 'black', margin: '8px' }}>
        <p style={{ color: '#fff', marginBottom: '2px' }}>
          <a href="https://goo.gl/maps/5G47ib81DGj7o2gk9" className="menu-hover" style={{ color: '#45c9ff', textDecoration: 'none' }}>
            <span>1601 Apperson Drive</span>
          </a>
          <br />
          Salem, VA 24153
        </p>
      </div>
    );
  }

  callText(): JSX.Element { // eslint-disable-line class-methods-use-this
    return (
      <a
        style={{
          display: 'block', color: '#fff', textDecoration: 'none', backgroundColor: '#881204', height: '72px', paddingTop: '4px', marginTop: '-1px',
        }}
        href="tel:5404447337"
      >
        <h4
          className="material-header-h4-call"
          style={{
            textAlign: 'center', marginRight: '280px', paddingBottom: 0, width: '200px',
          }}
        >
          Call
          <br />
          540-444-7337
        </h4>
      </a>
    );
  }

  mobileMenu(): JSX.Element {
    return (
      <div className="nav-list">
        {this.callText()}
        {this.addressBlock()}
        {this.menus.map((menu, index) => this.makeMenuLink(menu, index))}
      </div>
    );
  }

  navLinks(width: number): JSX.Element {
    if (width < 1162) return this.mobileMenu();
    return (
      <div className="nav-list">
        {this.addressBlock()}
        {this.menus.map((menu, index) => this.makeMenuLink(menu, index))}
      </div>
    );
  }

  headerSection(logoWidth: string, marginTop: string): JSX.Element { // eslint-disable-line class-methods-use-this
    return (
      <div id="header" className="material-header">
        <div className="headercontent" />
        <div>
          <div style={{ marginLeft: '5px', marginTop }}>
            <img
              width={logoWidth}
              id="ig"
              className="style-scope iron-image"
              alt="headerlogo"
              src="https://dl.dropboxusercontent.com/s/qg69q7f5so4aqsb/appAutoLogo.png?dl=0"
            />
          </div>
        </div>
      </div>
    );
  }

  callUs(): JSX.Element { // eslint-disable-line class-methods-use-this
    return (
      <div className="material-header x-scope paper-material-0 drawer" elevation="0" style={{ backgroundColor: '#881204' }}>
        {this.callText()}
      </div>
    );
  }

  render(): JSX.Element {
    let logoWidth = '742px', marginTop = '-15px';
    const { menuOpen } = this.state;
    const { targetRef, width } = this.props;
    if (width < 1232) { logoWidth = '272px'; marginTop = '1px'; }
    const style = `${menuOpen ? 'open' : 'close'}`;
    return (
      <div ref={targetRef} className="page-host">
        <div tabIndex={0} role="button" id="sidebar" onClick={this.close} onKeyPress={this.handleKeyPress} className={`${style} drawer-container`}>
          <div className="drawer" style={{ backgroundColor: '#505050' }}>
            {width > 1161 ? this.callUs() : null}
            { this.navLinks(width) }
          </div>
        </div>
        <div className="main-panel">
          <span onClick={this.toggleMobileMenu} onKeyPress={this.handleKeyMenu} id="mobilemenutoggle" tabIndex={0} role="button">
            <i className="fas fa-bars fa-2x" />
          </span>
          <div className="mainPanel">
            <div className="swipe-area" />
            {this.headerSection(logoWidth, marginTop)}
            <div style={{ width: 'auto' }} id="contentBlock" className="content-block">
              { this.children }
              <Footer />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

AppTemplate.defaultProps = {
  width: 320,
};

AppTemplate.propTypes = {
  location: PropTypes.shape({ pathname: PropTypes.string }).isRequired,
  children: PropTypes.element.isRequired,
  width: PropTypes.number,
  targetRef: PropTypes.shape({ current: PropTypes.instanceOf(Element) }).isRequired,
};

export default withRouter(connect(mapStoreToProps, null)(withResizeDetector(AppTemplate)));
