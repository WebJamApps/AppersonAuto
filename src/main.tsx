import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import { App } from './App';
import '../static/styles.scss';

const root = createRoot(document.getElementById('root') as HTMLElement);

const renderMain = (): void => {
  root.render(
    <StrictMode>
     <App />
    </StrictMode>,
  );
};

renderMain();
