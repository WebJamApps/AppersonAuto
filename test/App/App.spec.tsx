import renderer from 'react-test-renderer';
import { App } from '../../src/App';

describe('App component', () => {
  it('renders the component', () => {
    const result = renderer.create(<App />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
