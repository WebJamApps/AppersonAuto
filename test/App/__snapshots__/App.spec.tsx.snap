// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the component 1`] = `
<div
  className="App"
  id="App"
>
  <div
    className="page-host"
  >
    <div
      className="close drawer-container"
      id="sidebar"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="drawer"
        style={
          {
            "backgroundColor": "#505050",
          }
        }
      >
        <div
          className="nav-list"
        >
          <a
            href="tel:5404447337"
            style={
              {
                "backgroundColor": "#881204",
                "color": "#fff",
                "display": "block",
                "height": "72px",
                "marginTop": "-1px",
                "paddingTop": "4px",
                "textDecoration": "none",
              }
            }
          >
            <h4
              className="material-header-h4-call"
              style={
                {
                  "marginRight": "280px",
                  "paddingBottom": 0,
                  "textAlign": "center",
                  "width": "200px",
                }
              }
            >
              Call
              <br />
              540-444-7337
            </h4>
          </a>
          <div
            className="menu-item"
            style={
              {
                "backgroundColor": "black",
                "margin": "8px",
              }
            }
          >
            <p
              style={
                {
                  "color": "#fff",
                  "marginBottom": "2px",
                }
              }
            >
              <a
                className="menu-hover"
                href="https://goo.gl/maps/5G47ib81DGj7o2gk9"
                style={
                  {
                    "color": "#45c9ff",
                    "textDecoration": "none",
                  }
                }
              >
                <span>
                  1601 Apperson Drive
                </span>
              </a>
              <br />
              Salem, VA 24153
            </p>
          </div>
          <div
            className="menu-item"
          >
            <a
              className="nav-link"
              href="/auto-maintenance"
              onClick={[Function]}
            >
              <i
                className=""
              />
               
              <span
                className="nav-item"
              >
                Auto Maintenance
              </span>
            </a>
          </div>
          <div
            className="menu-item"
          >
            <a
              className="nav-link"
              href="/general-auto-repair"
              onClick={[Function]}
            >
              <i
                className=""
              />
               
              <span
                className="nav-item"
              >
                General Auto Repair
              </span>
            </a>
          </div>
          <div
            className="menu-item"
          >
            <a
              className="nav-link"
              href="/major-auto-repair"
              onClick={[Function]}
            >
              <i
                className=""
              />
               
              <span
                className="nav-item"
              >
                Major Auto Repair
              </span>
            </a>
          </div>
          <div
            className="menu-item"
          >
            <a
              className="nav-link"
              href="/contact"
              onClick={[Function]}
            >
              <i
                className=""
              />
               
              <span
                className="nav-item"
              >
                Contact
              </span>
            </a>
          </div>
          <div
            className="menu-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              <i
                className=""
              />
               
              <span
                className="nav-item"
              >
                Home
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="main-panel"
    >
      <span
        id="mobilemenutoggle"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          className="fas fa-bars fa-2x"
        />
      </span>
      <div
        className="mainPanel"
      >
        <div
          className="swipe-area"
        />
        <div
          className="material-header"
          id="header"
        >
          <div
            className="headercontent"
          />
          <div>
            <div
              style={
                {
                  "marginLeft": "5px",
                  "marginTop": "-15px",
                }
              }
            >
              <img
                alt="headerlogo"
                className="style-scope iron-image"
                id="ig"
                src="https://dl.dropboxusercontent.com/s/qg69q7f5so4aqsb/appAutoLogo.png?dl=0"
                width="742px"
              />
            </div>
          </div>
        </div>
        <div
          className="content-block"
          id="contentBlock"
          style={
            {
              "width": "auto",
            }
          }
        >
          <div>
            <div
              className="container-fluid"
            >
              <span
                className="top"
              />
              <div
                className="row"
              >
                <div
                  className="col"
                  style={
                    {
                      "marginLeft": "15px",
                      "paddingRight": "6px",
                      "top": "0",
                    }
                  }
                >
                  <h4
                    className="pagetitle"
                    style={
                      {
                        "fontSize": "16pt",
                        "fontWeight": "bold",
                        "marginTop": "25px",
                      }
                    }
                  >
                    Your one stop auto shop where honesty, quality work, and fair pricing are guaranteed
                  </h4>
                  <p>
                     
                  </p>
                  <div>
                    <p>
                      Are you looking for well qualified mechanics? Look no further!
                      <br />
                      We have more than
                       
                      <strong>
                        150 years of combined experience
                      </strong>
                      !
                    </p>
                    <p>
                      Take advantage of our
                      <br />
                      <strong>
                        FREE high mileage / vacation vehicle check over
                      </strong>
                      .
                    </p>
                  </div>
                  <h4
                    style={
                      {
                        "marginTop": "40px",
                      }
                    }
                  >
                    Apperson Automotive - Salem, VA Auto Repair Service
                  </h4>
                  <p>
                    <strong>
                      Quality service for your vehicle (not limited to):
                    </strong>
                  </p>
                  <ul>
                    <li>
                      State inspections
                    </li>
                    <li>
                      Alignments
                    </li>
                    <li>
                      Tune ups
                    </li>
                    <li>
                      Oil changes
                    </li>
                    <li>
                      Coolant flush
                    </li>
                  </ul>
                  <div>
                    <p>
                      <strong>
                        Get to Know Apperson Automotive:
                      </strong>
                    </p>
                    <ul>
                      <li>
                        <a
                          href="/auto-maintenance"
                          title="Apperson Automotive Auto Maintenance"
                        >
                          Auto Maintenance
                        </a>
                         
                        – Let us handle your oil change, coolant flushes, alignments, and more
                      </li>
                      <li>
                        <a
                          href="/general-auto-repair"
                          title="Apperson Automotive General Auto Repair"
                        >
                          General Auto Repair
                        </a>
                         
                        – General repairs for includes (not limited to) brake calipers, window motors, regulators, and starters
                      </li>
                      <li>
                        <a
                          href="/major-auto-repair"
                          title="Apperson Automotive Major Auto Repair"
                        >
                          Major Auto Repair
                        </a>
                         
                        – We’ll handle your steering rack issues, intake gasket repairs, heat and air system problems, and much more
                      </li>
                    </ul>
                    <p>
                      We’ll repair any make or model vehicle you have.
                    </p>
                    <p>
                      To schedule your FREE high mileage or vacation vehicle check over, call Apperson Automotive at
                       
                      <a
                        href="tel:5404447337"
                        style={
                          {
                            "textDecoration": "none",
                          }
                        }
                      >
                        540-444-7337
                      </a>
                      <br />
                      You can also stop by at 1601 Apperson Drive, Salem VA 24153.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="footer"
            id="wjfooter"
            style={
              {
                "bottom": "0",
                "marginTop": "20px",
                "paddingBottom": "20px",
                "paddingTop": "20px",
              }
            }
          >
            <div
              className="footerLinks"
              style={
                {
                  "padding": "6px",
                  "textAlign": "center",
                }
              }
            >
              <a
                href="https://www.facebook.com/pages/Apperson-Automotive/361511727202422"
              >
                <i
                  aria-hidden="true"
                  className="fab fa-facebook-square fa-2x"
                  style={
                    {
                      "color": "#45c9ff",
                    }
                  }
                />
              </a>
              <a
                href="https://www.yellowpages.com/salem-va/mip/apperson-automotive-482421530"
                style={
                  {
                    "marginLeft": "40px",
                  }
                }
              >
                <img
                  alt="yellowpages"
                  src="https://insightmg.com/wp-content/uploads/2015/12/YP_Insight-Marketing-Group_Marketing-for-Medical-Practices.png"
                  style={
                    {
                      "marginBottom": "15px",
                      "verticalAlign": "middle",
                    }
                  }
                  width="40px"
                />
              </a>
              <a
                href="https://www.yelp.com/biz/apperson-automotive-salem"
                style={
                  {
                    "marginLeft": "30px",
                  }
                }
              >
                <img
                  alt="yelp"
                  src="https://dl.dropboxusercontent.com/s/pb5qn5wyps6k56t/yelp-logo-icon-7.jpg?dl=0"
                  style={
                    {
                      "marginBottom": "15px",
                      "verticalAlign": "middle",
                    }
                  }
                  width="40px"
                />
              </a>
            </div>
            <div
              style={
                {
                  "padding": "6px",
                  "textAlign": "center",
                }
              }
            >
              <span
                className="nohover"
              >
                Powered by  
              </span>
              <a
                className="wjllc"
                href="https://www.web-jam.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Web Jam LLC
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
